activerecord/lib/active_record/core.rb

```diff
@@ -155,6 +155,19 @@ def self.configurations
 
       mattr_accessor :legacy_connection_handling, instance_writer: false, default: true
 
+      mattr_accessor :application_record_class, instance_accessor: false, default: nil
+
+      def self.application_record_class? # :nodoc:
+        if Base.application_record_class
+          self == Base.application_record_class
+        else
+          if defined?(ApplicationRecord) && self == ApplicationRecord
```

Comment:
This is here only for backward compatibility right? Should we deprecate?

Reply:
It's less backwards compatibility and more "most likely to be the default". I was thinking that I didn't want to force any existing application using `ApplicationRecord` to have to change anything. We could update the generators to add `primary_abstract_class` to `ApplicationRecord` and deprecate this to get everyone on board, but it only matters in apps that are using multiple databases AND have a non-application record application record. I don't think the number of apps doing this is high enough to warrant everyone changing their existing applications. What do you think?

Reply:
Agree with what you said. I'd still update the generator to use `primary_abstract_class`.